src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,209 @@
+            throw new InvalidArgumentError('Unable to submit payment because "payload.payment" argument is not provided.');
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paymentInitData = this._paymentMethod.initializationData;
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({
+                    ...payment, paymentData: {
+                        cryptogramId: paymentInitData.paymentCryptogram,
+                        eci: paymentInitData.eci,
+                        transactionId: btoa(paymentInitData.reqTokenId),
+                        ccExpiry: {
+                            month: paymentInitData.expDate.toString().substr(0, 2),
+                            year: paymentInitData.expDate.toString().substr(2, 2),
+                        },
+                        ccNumber: paymentInitData.accountNum,
+                        accountMask: paymentInitData.accountMask,
+                        extraData: { riskToken: this.riskToken },
+                    },
+                }))
+            );
+    }
+
+    private _paymentInstrumentSelected() {
+        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
+            return Promise.all([
+                this._store.dispatch(this._checkoutActionCreator.loadCurrentCheckout()),
+                this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod('chasepay')),
+            ]);
+        }, { methodId: 'chasepay' }), { queueId: 'widgetInteraction' });
+    }
+
+    private _refreshDigitalSessionId(methodId: string): Promise<string | undefined> {
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+                if (!this._paymentMethod) {
+                    return;
+                }
+                return this._paymentMethod.initializationData.digitalSessionId;
+            });
+    }
+
+    private _setExternalCheckoutData(payload: ChasePaySuccessPayload, requestId: string): Promise<Response> {
+        const url = `checkout.php?provider=chasepay&action=set_external_checkout`;
+        const options = {
+            headers: {
+                Accept: 'text/html',
+                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+            },
+            body: toFormUrlEncoded({
+                sessionToken: payload.sessionToken,
+                merchantRequestId: requestId,
+            }),
+        };
+
+        return this._requestSender.post(url, options);
+    }
+}
+
+export interface ChasePayInitializeOptions {
+    container: string;
```

Comment:
Do you have a description for this? This seems to be just for inserting a logo. So it should be optional. It should also be renamed to something more appropriate.